rest/server.go

```diff
@@ -0,0 +1,108 @@
+package rest
+
+import (
+	"awesomeProject3/service"
+	"context"
+	"encoding/json"
+	"github.com/go-chi/chi/v5"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+type Handler func(w http.ResponseWriter, r *http.Request) error
+
+func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if err := h(w, r); err != nil {
+		w.WriteHeader(503)
+	}
+}
+
+type container struct {
+	Getter string `json:"getter"`
+	Value  string `json:"value"`
+}
+
+type SecretRestAPI struct {
+	ssService  service.SecretService
+	httpServer *http.Server
+}
+
+func NewSecretRestAPI(sr service.SecretService, portnumber string) *SecretRestAPI {
+	a := SecretRestAPI{}
+
+	h := chi.NewRouter()
+	h.Method("GET", "/", Handler(a.handlerGet))
+	h.Method("POST", "/", Handler(a.handlerPost))
+
+	a.ssService = sr
+	a.httpServer = &http.Server{Addr: "localhost:" + portnumber, Handler: h}
+
+	return &a
+}
+
+func (sr *SecretRestAPI) Start() {
+	serverCtx, serverStopCtx := context.WithCancel(context.Background())
+
+	sign := make(chan os.Signal, 1)
+	signal.Notify(sign, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+	go func() {
+		<-sign
```

Comment:
What if `serverCtx` get canceled earlier that any signal comes from `sign` channel? How to manage reads from  several channels?